### PR TITLE
escape spaces in git submodule path

### PR DESCRIPTION
**Pull request by kashif**
for issue #36646

### CODE REVIEW

File: Library/Homebrew/download_strategy.rb [hunk 1]
```diff
@@ -652,7 +652,7 @@ def update_submodules
 
   def checkout_submodules(dst)
     escaped_clone_path = cached_location.to_s.gsub(/\//, '\/')
-    sub_cmd = "git checkout-index -a -f --prefix=#{dst}/${toplevel/#{escaped_clone_path}/}/$path/"
+    sub_cmd = "git checkout-index -a -f --prefix=#{dst}/${toplevel/#{escaped_clone_path}/}/\'$path\'/"
```
Comment (jacknagel) [inline Library/Homebrew/download_strategy.rb]: Do the single quotes actually need to be escaped?

```rb
irb(main):001:0> "\'"
=> "'"
irb(main):002:0> "'"
=> "'"
```

Comment (kashif) [inline Library/Homebrew/download_strategy.rb]: ah yes... i was also thinking about using `shellescape` from the ruby stdlib...

Comment (jacknagel) [inline Library/Homebrew/download_strategy.rb]: I don't think we want to shell escape this. This string is passed directly to git, and git eventually passes it to the shell where `$path` and `$toplevel` are populated.

I was just pointing out that the `\` is not strictly necessary here.

Comment (kashif) [inline Library/Homebrew/download_strategy.rb]: @jacknagel ok i'll make a pull request in a few hours... just in the middle of a meeting

Comment (jacknagel) [inline Library/Homebrew/download_strategy.rb]: No worries, I just wanted to be sure I understood the intent correctly.

**Pull request closed without merging**